### PR TITLE
refactor(Islands): report card part of build step

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

Refactor Islands report card, so that it gets deployed to production.

Small refactor to make internal methods purely functional, and invoke the script only once at the very end.

## Why?

Introduced in #7124, this process did not produce the assets to be uploaded to S3

### CODE REVIEW

File: dotcom-rendering/makefile [hunk 1]
```diff
@@ -33,6 +33,8 @@ deploy:
 build: clean-dist install
 	$(call log, "building production bundles")
 	@NODE_ENV=production webpack --config ./scripts/webpack/webpack.config.js --progress
+	$(call log, "generating Islands report card")
+	@node ./cripts/islands/island-descriptions.mjs
```
Comment (OllysCoding) [inline dotcom-rendering/makefile:37]:
```suggestion
	@node ./scripts/islands/island-descriptions.mjs
```

File: dotcom-rendering/scripts/islands/island-descriptions.mjs [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+		resolve(
+			fileURLToPath(import.meta.url),
+			'../../../dist/browser.modern-bundles.html',
+		),
+		'utf-8',
+	)
+		.then((file) =>
+			file
+				.split('\n')
+				.find((line) => line.trim().startsWith(PREFIX))
```
Comment (OllysCoding) [inline dotcom-rendering/scripts/islands/island-descriptions.mjs:51]: I think we can avoid this ... awkwardness ... by using the `generateStatsFile` option for webpack bundle analyser? https://github.com/webpack-contrib/webpack-bundle-analyzer#options-for-plugin

Comment (sndrs) [reply dotcom-rendering/scripts/islands/island-descriptions.mjs:51]: is that file a problem? how much of an impact does it have?

Comment (mxdvl) [reply dotcom-rendering/scripts/islands/island-descriptions.mjs:51]: I think that the server toppled over last year from having too many files on disk, hence my reservation, as it’s Webpack’s full bundle JSON, which does not include gzipped sizes.

I could try running the WebpackAnalyser plugin twice, once with the `html` output and once with the `json` output, possibly?

File: dotcom-rendering/scripts/islands/island-descriptions.mjs [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+	.use(rehypeSlug)
+	.use(rehypeAutolinkHeadings, { behavior: 'wrap', test: 'h1' })
+	.use(rehypeStringify);
+
+/**
+ * @param {Awaited<ReturnType<typeof getBundleReport>>} report
+ */
+const getIslands = (report) =>
+	readdir(componentsDirectory)
+		.then((components) =>
```
Comment (sndrs) [inline dotcom-rendering/scripts/islands/island-descriptions.mjs:73]: more of a style thing, but all these `then`s are hard to follow. you have actually go in and read the code to know what each one does. breaking them down into `const`s with `await`ed values might help that a bit?

**Review by sndrs — APPROVED** [review]
aside from my comment about legibility of multiple, large `then`s, great! (your call)
